packages/webapp/src/lib/services/handleVaultsWithdrawAll.ts

```diff
@@ -0,0 +1,94 @@
+import type { RaindexClient, RaindexVault, VaultsList } from '@rainlanguage/orderbook';
+import { type Hex } from 'viem';
+import { QKEY_VAULTS, type TransactionManager } from '@rainlanguage/ui-components';
+import type { TransactionConfirmationProps } from '@rainlanguage/ui-components';
+import type { TransactionConfirmationModalResult } from './modal';
+
+export type WithdrawAllModalProps = {
+	open: boolean;
+	vaults: RaindexVault[];
+	onSubmit: () => void;
+};
+
+export interface MultipleVaultsWithdrawHandlerDependencies {
+	raindexClient: RaindexClient;
+	vaultsList: VaultsList;
+	handleWithdrawAllModal: (props: WithdrawAllModalProps) => void;
+	handleTransactionConfirmationModal: (
+		props: TransactionConfirmationProps
+	) => Promise<TransactionConfirmationModalResult>;
+	errToast: (message: string) => void;
+	manager: TransactionManager;
+	account: Hex;
+}
+
+export async function handleVaultsWithdrawAll(
+	deps: MultipleVaultsWithdrawHandlerDependencies
+): Promise<boolean> {
+	const {
+		raindexClient,
+		vaultsList,
+		handleWithdrawAllModal,
+		handleTransactionConfirmationModal,
+		errToast,
+		manager
+	} = deps;
+	const vaultsResult = vaultsList.getWithdrawableVaults();
+	if (vaultsResult.error) {
+		errToast(`Failed to get withdrawable vaults: ${vaultsResult.error.readableMsg}`);
+		return false;
+	}
+	const vaults = vaultsResult.value;
+	// Early return if no vaults are selected
+	if (vaults.length === 0) {
+		errToast('No vaults selected for withdrawal.');
+		return false;
+	}
+
+	return new Promise((resolve) => {
+		handleWithdrawAllModal({
+			open: true,
+			vaults,
+			onSubmit: async () => {
+				try {
+					// Validate that all vaults share the same orderbook
+					const orderbook = vaults[0].orderbook;
```

Comment:
i feel like this logic can be done inside `getWithdrawableVaults`

so we don't have to call it here, we can just call `getWithdrawCalldata`